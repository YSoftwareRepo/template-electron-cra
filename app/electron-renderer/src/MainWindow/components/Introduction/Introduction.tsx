import Assets from '@tecra-pkg/assets';
import type { AppDetails } from '@tecra-pkg/electron-common';

import css from './Introduction.module.scss';

export interface IntroductionProps {
    appDetails: AppDetails;
}

export const Introduction: React.FC<IntroductionProps> = ({ appDetails }) => (
    <>
        <header className={css.introduction}>
            <img className={css.logo} src={Assets.logo} alt="logo" />
            <p>
                Edit <code>src/MainWindow/components/Introduction/Introduction.tsx</code> and save to reload.
            </p>
            <a
                className={css.link}
                href="https://github.com/YSoftwareRepo/template-electron-cra"
                target="_blank"
                rel="noopener noreferrer"
            >
                Learn template-electron-cra
            </a>
        </header>
        <div className={css.appDetails}>
            <code>Name: {appDetails.name}</code>
            <code>Version: {appDetails.version}</code>
            <code>Electron: {appDetails.module.electron}</code>
            <code>Chrome: {appDetails.module.chrome}</code>
            <code>Node.js: {appDetails.module.node}</code>
            <code>V8: {appDetails.module.v8}</code>
        </div>
    </>
);
