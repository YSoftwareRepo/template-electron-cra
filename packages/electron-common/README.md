# @tecra-pkg/electron-common

Common code for electron main process and electron renderer process.
