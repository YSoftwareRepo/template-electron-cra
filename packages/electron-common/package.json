{
    "name": "@tecra/electron-common",
    "version": "0.8.0",
    "description": "Common code for electron main process and electron renderer process.",
    "main": "lib/index.js",
    "module": "lib/index.js",
    "typings": "lib/index.d.ts",
    "homepage": "https://github.com/YSoftwareRepo/template-electron-cra#readme",
    "license": "MIT",
    "author": "yifanwww <yifanw1101@gmail.com> (https://github.com/yifanwww)",
    "bugs": {
        "url": "https://github.com/YSoftwareRepo/template-electron-cra/issues"
    },
    "private": true,
    "repository": {
        "type": "git",
        "url": "git+https://github.com/YSoftwareRepo/template-electron-cra.git",
        "directory": "packages/electron-common"
    },
    "scripts": {
        "build": "concurrently -n esm,types \"pnpm run build:esm\" \"pnpm run build:types\"",
        "build:esm": "tsc --project tsconfigs/tsconfig.esm.json",
        "build:types": "tsc --project tsconfigs/tsconfig.types.json",
        "clean": "rimraf coverage lib tsconfigs/*.tsbuildinfo",
        "dev": "pnpm run build:esm -- --watch",
        "format": "format-code",
        "lint": "eslint .",
        "pre-commit": "lint-staged"
    },
    "devDependencies": {
        "@tecra/eslint-config": "workspace:*",
        "@tecra/global-types": "workspace:*",
        "@tecra/scripts": "workspace:*",
        "@tecra/tsconfigs": "workspace:*"
    }
}
