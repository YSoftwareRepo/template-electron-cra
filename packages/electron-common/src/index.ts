export * from './apis';
export * from './timer';
export * from './type';
export * from './utils';
