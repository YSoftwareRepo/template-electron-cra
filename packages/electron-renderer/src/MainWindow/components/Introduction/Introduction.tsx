import { Assets } from '@tecra/assets';

import scss from './Introduction.module.scss';

export function Introduction(): React.ReactElement {
    return (
        <header className={scss.introduction}>
            <img className={scss.logo} src={Assets.logo} alt="logo" />
            <p>
                Edit <code>src/MainWindow/components/Introduction/Introduction.tsx</code> and save to reload.
            </p>
            <a
                className={scss.link}
                href="https://github.com/YSoftwareRepo/template-electron-cra"
                target="_blank"
                rel="noopener noreferrer"
            >
                Learn template-electron-cra
            </a>
        </header>
    );
}
