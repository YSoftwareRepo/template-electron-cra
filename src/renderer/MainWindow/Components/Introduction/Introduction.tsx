import { keyframes, makeStyles } from '@fluentui/react';
import { ReactElement } from 'react';

import logo from './logo.svg';

const useClassNames = makeStyles(() => {
    const _keyframes = keyframes({
        from: { transform: 'rotate(0deg)' },
        to: { transform: 'rotate(360deg)' },
    });

    return {
        introduction: {
            alignItems: 'center',
            background: '#282c34',
            color: 'white',
            display: 'flex',
            flexDirection: 'column',
            fontSize: 'calc(10px + 2vmin)',
            justifyContent: 'center',
            minHeight: '100vh',
        },
        link: {
            color: '#61dafb',
        },
        logo: {
            height: '40vmin',
            pointerEvents: 'none',

            '@media (prefers-reduced-motion: no-preference)': {
                animation: `${_keyframes} infinite 20s linear`,
            },
        },
    };
});

export function Introduction(): ReactElement {
    const classNames = useClassNames();

    return (
        <header className={classNames.introduction}>
            <img className={classNames.logo} src={logo} alt="logo" />
            <p>
                Edit <code>src/renderer/MainWindow/App.tsx</code> and save to reload.
            </p>
            <a
                className={classNames.link}
                href="https://github.com/YSoftwareRepo/template-electron-cra"
                target="_blank"
                rel="noopener noreferrer"
            >
                Learn @YSoftwareRepo/template-electron-cra
            </a>
        </header>
    );
}
