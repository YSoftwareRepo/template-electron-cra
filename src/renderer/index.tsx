import { ReactElement, StrictMode } from 'react';
import ReactDOM from 'react-dom';
import { initializeIcons, ThemeProvider } from '@fluentui/react';

import { WindowType } from '#shared/WindowType';
import { windowIpc } from '#RUtils/Ipc';
import { fluentuiTheme } from '#RUtils/Theme';

import { MainWindow } from './MainWindow';

import './index.css';
import * as serviceWorker from './serviceWorker';

function WindowSwitch(): ReactElement {
    const windowType: WindowType = windowIpc.getWindowType();

    // eslint-disable-next-line default-case
    switch (windowType) {
        case 'main':
            return <MainWindow />;
    }
}

initializeIcons();

ReactDOM.render(
    <StrictMode>
        <ThemeProvider id="ThemeProvider" theme={fluentuiTheme}>
            <WindowSwitch />
        </ThemeProvider>
    </StrictMode>,
    document.getElementById('root'),
);

// If you want your app to work offline and load faster, you can change unregister() to register() below. Note this
// comes with some pitfalls.
// Learn more about service workers: https://bit.ly/CRA-PWA
serviceWorker.unregister();
