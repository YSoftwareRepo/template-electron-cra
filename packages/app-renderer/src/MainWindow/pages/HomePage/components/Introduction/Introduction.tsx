import type { AppDetails, Nullable } from '@ter/app-common';

import { Assets } from 'src/assets';

import css from './Introduction.module.scss';

interface IntroductionProps {
    appDetails: Nullable<AppDetails>;
}

export function Introduction({ appDetails }: IntroductionProps): JSX.Element {
    return (
        <>
            <header className={css.introduction}>
                <img className={css.logo} src={Assets.logo} alt="logo" />
                <p>
                    Edit <code>src/MainWindow/components/Introduction/Introduction.tsx</code> and save to reload.
                </p>
                <a
                    className={css.link}
                    href="https://github.com/YSoftwareRepo/template-electron-react"
                    target="_blank"
                    rel="noopener noreferrer"
                >
                    Learn template-electron-react
                </a>
            </header>
            {appDetails && (
                <div className={css.appDetails}>
                    <code>Name: {appDetails.name}</code>
                    <code>Version: {appDetails.version}</code>
                    <code>Electron: {appDetails.module.electron}</code>
                    <code>Chrome: {appDetails.module.chrome}</code>
                    <code>Node.js: {appDetails.module.node}</code>
                    <code>V8: {appDetails.module.v8}</code>
                </div>
            )}
        </>
    );
}
