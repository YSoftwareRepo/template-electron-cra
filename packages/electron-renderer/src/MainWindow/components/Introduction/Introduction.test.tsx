import { AppDetails } from '@tecra/electron-common';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Introduction } from './Introduction';

describe('Test component `Introduction`', () => {
    it('renders', () => {
        const appDetails: AppDetails = {
            name: 'tecra',
            version: 'unknown',
            module: {
                chrome: 'unknown',
                electron: 'unknown',
                node: 'unknown',
                v8: 'unknown',
            },
        };

        const reactElement = <Introduction appDetails={appDetails} />;

        const component = renderer.create(reactElement);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        const { getByText } = render(reactElement);
        const linkElement = getByText(/Learn template-electron-cra/i);
        expect(linkElement).toBeInTheDocument();
    });
});
