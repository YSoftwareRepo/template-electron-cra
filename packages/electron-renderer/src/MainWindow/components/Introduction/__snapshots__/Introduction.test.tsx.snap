// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component \`Introduction\` renders 1`] = `
<header
  className="introduction"
>
  <img
    alt="logo"
    className="logo"
    src="logo.svg"
  />
  <p>
    Edit 
    <code>
      src/MainWindow/components/Introduction/Introduction.tsx
    </code>
     and save to reload.
  </p>
  <a
    className="link"
    href="https://github.com/YSoftwareRepo/template-electron-cra"
    rel="noopener noreferrer"
    target="_blank"
  >
    Learn template-electron-cra
  </a>
</header>
`;
